src/vcon/MarketGovernance.sol

```diff
@@ -0,0 +1,653 @@
+
+    /// @notice update the venue last recorded profit
+    /// and the venue last recorded vcon share price
+    /// system must be at lock level 1 to call this function,
+    /// otherwise call to `accrue()` on the PCV Deposit will fail
+    /// @dev this function assumes the venue is in the PCV Oracle as the
+    /// calling function should either check, or be callable only by
+    /// governance
+    function _accrue(address venue) private {
+        /// cache starting recorded profit before the external call even though
+        /// there is no way to call _accrue without setting the global reentrancy lock to level 1
+        uint256 startingLastRecordedProfit = venueLastRecordedProfit[venue];
+
+        IPCVDepositV2(venue).accrue();
+
+        uint256 lastRecordedProfit = IPCVDepositV2(venue).lastRecordedProfit();
+        uint256 endingLastRecordedSharePrice = venueLastRecordedVconSharePrice[
+            venue
+        ];
+        int256 venueProfit = (lastRecordedProfit.toInt256() -
+            startingLastRecordedProfit.toInt256());
+
+        require(venueProfit >= 0, "MarketGovernance: loss scenario");
+
+        /// update venue last recorded profit regardless
+        /// of participation in market governance
+        if (endingLastRecordedSharePrice != 0) {
+            uint256 venueShares = venueTotalShares[venue];
+
+            /// if venue has 0 staked vcon, do not update share price, just update profit index
+            if (venueShares != 0) {
+                uint256 venueProfitRatio = profitToVconRatio[venue];
+
+                uint256 vconEarnedPerShare = (Constants.ETH_GRANULARITY *
+                    venueProfit.toUint256() *
+                    venueProfitRatio) / venueShares;
+
+                /// gain scenario
+                venueLastRecordedVconSharePrice[venue] += vconEarnedPerShare
+                    .toUint128();
+            }
+        } else {
+            /// share price is 0, and requires initialization
+            venueLastRecordedVconSharePrice[venue] = Constants
+                .ETH_GRANULARITY
+                .toUint128();
+        }
+
+        /// update the venue's profit index
+        venueLastRecordedProfit[venue] = lastRecordedProfit.toUint128();
+
+        emit VenueIndexUpdated(venue, block.timestamp, lastRecordedProfit);
+    }
+
+    /// ---------- Governor-Only Permissioned API ----------
+
+    /// @notice governor only function to set the profit to VCON ratio
+    /// this function will not be callable if an underlying venue took a loss as `_accrue()`
+    /// will revert and no users will be able to withdraw their VCON.
+    function setProfitToVconRatio(
+        address venue,
+        uint256 newProfitToVconRatio
+    ) external onlyGovernor globalLock(1) {
+        /// lock to level 1 so that accrue can succeed
+        require(pcvOracle().isVenue(venue), "MarketGovernance: invalid venue");
+        _accrue(venue); /// ensure users receive all rewards from the old rate
```

Comment:
you could not lock, not do the check, and just use `systemEntry.accrue(venue)`

Reply:
and then we'd have yet another reference to manage in this contract.

Reply:
okok